Update VSCode integration (#12068)
godotengine/godot-docs: tutorials/editor/external_editor.rst

| JetBrains Rider     | ``{project} --line {line} {file}``                                 |
+---------------------+--------------------------------------------------------------------+
| Visual Studio Code  | ``--reuse-window {project} --goto {file}:{line}:{col}``            |
+---------------------+--------------------------------------------------------------------+
| Vim (gVim)          | ``"+call cursor({line}, {col})" {file}``                           |
+---------------------+--------------------------------------------------------------------+
| Emacs               | ``emacs +{line}:{col} {file}``                                     |
+---------------------+--------------------------------------------------------------------+
| Sublime Text/Zed    | ``{project} {file}:{line}:{col}``                                  |
+---------------------+--------------------------------------------------------------------+
| Visual Studio*      | ``/edit "{file}"``                                                 |
+---------------------+--------------------------------------------------------------------+

> 
> Some examples of **Exec Path** include:
> 
> +-----------------+---------------------+---------------------------------------------+
> | Godot version   | Exec Path           | Exec Flags                                  |
> +=================+=====================+=============================================+
> | Flatpak         | ``flatpak-spawn``   | ``--host <Editor Path> <Exec Flags>``       |
> +-----------------+---------------------+---------------------------------------------+

Reviewer: In my opinion the wording and placement is a little confusing. I think a separate admonition makes it clearer, something like this:

```suggestion
.. note::

    If you-re using the Flatpak version of Godot, you must instead use ``flatpak-spawn`` to invoke your editor of choice:
    
    - **Exec Path**: ``flatpak-spawn``
    - **Exec Flags**: ``-host <Editor Path> <Exec Flags>``
```

You need to install the official `Visual Studio Code plugin <https://github.com/godotengine/godot-vscode-plugin>`_.

By default, the path to the editor executable is ``godot`` but you can assign a different path to the Setting ID ``godotTools.editorPath.godot4``. Some examples include:

+-----------------+-----------------------------------------+
| Godot version   | Editor Path                             |
+=================+=========================================+
| Flatpak         | ``flatpak run org.godotengine.Godot``   |
+-----------------+-----------------------------------------+

> For **LSP**, set both Godot's Editor Setting ``network/language_server/remote_port`` and VSCode Setting ID ``godotTools.lsp.serverPort`` to change the default LSP port. The connection status can be checked on the status bar:

Reviewer: No need to reference the Godot's editor setting `network/language_server/remote_port`, as it's already documented on the general section, common to all editor examples:

```suggestion
For **LSP**, if not using the default Godot LSP port, you need to set VSCode's Setting ID ``godotTools.lsp.serverPort`` to the correct port. The connection status can be checked on the status bar:
```

Reviewer 2: I think the wording here should indicate, that out-of-the-box you don't need to configure the port. This is only really relevant if you changed the setting in Godot.

Reviewer: Great point, in fact, that was the implicit meaning before these changes. I've amended my suggested edit with this.
